modular_skyrat/modules/goofsec/code/sol_fed.dm

```diff
@@ -1,8 +1,236 @@
+
+	if(candidates.len)
+		//Pick the (un)lucky players
+		var/agents_number = min(team_size,candidates.len)
+
+		var/list/spawnpoints = GLOB.emergencyresponseteamspawn
+		var/index = 0
+		GLOB.solfed_responder_info[list_to_use][SOLFED_AMT] = agents_number
+		while(agents_number && candidates.len)
+			var/spawnloc = spawnpoints[index+1]
+			//loop through spawnpoints one at a time
+			index = (index + 1) % spawnpoints.len
+			var/mob/dead/observer/chosen_candidate = pick(candidates)
+			candidates -= chosen_candidate
+			if(!chosen_candidate.key)
+				continue
+
+			//Spawn the body
+			var/mob/living/carbon/human/cop = new(spawnloc)
+			chosen_candidate.client.prefs.safe_transfer_prefs_to(cop, is_antag = TRUE)
+			cop.key = chosen_candidate.key
+
+			//Give antag datum
+			var/datum/antagonist/ert/request_911/ert_antag = new cops_to_send
+
+			cop.mind.add_antag_datum(ert_antag)
+			cop.mind.set_assigned_role(SSjob.GetJobType(ert_antag.ert_job_path))
+			SSjob.SendToLateJoin(cop)
+			cop.grant_language(/datum/language/common, TRUE, TRUE, LANGUAGE_MIND)
+
+			var/obj/item/gangster_cellphone/phone = new() // biggest gang in the city
+			phone.gang_id = cell_phone_number
+			phone.name = "[cell_phone_number] branded cell phone"
+			var/phone_equipped = phone.equip_to_best_slot(cop)
+			if(!phone_equipped)
+				to_chat(cop, "Your [phone.name] has been placed at your feet.")
+				phone.forceMove(get_turf(cop))
+
+			//Logging and cleanup
+			log_game("[key_name(cop)] has been selected as an [ert_antag.name]")
+			agents_number--
+	GLOB.cops_arrived = TRUE
+	return TRUE
+
+/obj/machinery/computer/communications/proc/pre_911_check(mob/user)
+	if (!authenticated_as_silicon_or_captain(user))
+		return FALSE
+
+	if (GLOB.cops_arrived)
+		to_chat(user, span_warning("911 has already been called this shift!"))
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, FALSE)
+		return FALSE
+
+	if (!issilicon(user))
+		var/obj/item/held_item = user.get_active_held_item()
+		var/obj/item/card/id/id_card = held_item?.GetID()
+		if (!istype(id_card))
+			to_chat(user, span_warning("You need to swipe your ID!"))
+			playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, FALSE)
+			return FALSE
+		if (!(ACCESS_CAPTAIN in id_card.access))
+			to_chat(user, span_warning("You are not authorized to do this!"))
+			playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, FALSE)
+			return FALSE
+	else
+		to_chat(user, "The console refuses to let you dial 911 as an AI or Cyborg!")
+		return FALSE
+	return TRUE
+
+/obj/machinery/computer/communications/proc/calling_911(mob/user, called_group_pretty = "EMTs", called_group = EMERGENCY_RESPONSE_EMT)
+	message_admins("[ADMIN_LOOKUPFLW(user)] is considering calling the Sol Federation [called_group_pretty].")
+	if(tgui_input_list(user, "Are you sure you want to call 911? Faulty 911 calls results in a $20,000 fine and a 5 year superjail \
```

Comment:
put the text in a var to reduce the clutter here, multi-line if statements are counter-intuitive.